app/forms/form.rb

```diff
@@ -49,19 +53,28 @@ def persisted?
   private
 
   def permitted_attributes
-    attribute_names
+    attributes.keys.map do |key|
+      field = @attributes[key]
+
+      case field.value_before_type_cast
+      when []
+        {key => []}
+      else
+        key
+      end
+    end
   end
 
   def i18n_form_namespace
     self.class.name.demodulize.gsub("Form", "").underscore
   end
 
   def attributes_with_current_value
-    attributes.each_with_object({}) do |(attribute, _), attributes|
-      attributes[attribute] = permitted_params[attribute]
-      next unless attributes[attribute].nil?
+    attributes.each_with_object({}) do |(attribute, _), hash|
+      hash[attribute] = permitted_params[attribute]
+      next unless hash[attribute].nil?
 
-      attributes[attribute] = load_current_value(attribute)
+      hash[attribute] = load_current_value(attribute)
```

Comment:
no change. change local variable name to `hash` to disambiguate from `attributes` which is a method on the object